### PR TITLE
Improve handling of narrow screens

### PR DESCRIPTION
**Pull request by DanTup**
It tweaks some existing CSS we had to collapse the inspector buttons, making them a bit more generic (using the widths to collapse at in the class names - I can't think of a better way to handle this?).

Here's some screenshots resizing down to 500px (getting below 500px will mean doing something more than just trimming things down, like scrolling the nav of having it expand/collapse).

Screenshot 2019-03-19 at 5 39 05 pm
---
Screenshot 2019-03-19 at 5 39 21 pm
---
Screenshot 2019-03-19 at 5 39 37 pm
---
Screenshot 2019-03-19 at 5 39 53 pm
---
Screenshot 2019-03-19 at 5 40 16 pm

### CODE REVIEW

**Comment by DanTup** [comment]
I changed this to hide text instead of icons, which makes it must easier to fit at 400 (and ditch the whole title together). This cleans up some of the CSS too.

Let me know what you think.

Screenshot 2019-03-19 at 6 01 15 pm
---
Screenshot 2019-03-19 at 6 01 27 pm
---
Screenshot 2019-03-19 at 6 01 36 pm
---
Screenshot 2019-03-19 at 6 01 53 pm

**Comment by DanTup** [comment]
Oh, forgot to screenshot the connect dialog - it shouldn't wrap anymore (primer has a max-width of 100% and a width of 440px - I changed it to be a `calc()` to remove the width of the button).

Screenshot 2019-03-19 at 6 06 20 pm

**Comment by DanTup** [comment]
@devoncarew @jacob314 I've pushed some additional changes that:

- collapse memory buttons for narrow windows
- Changes logging view from horizontal to vertical for narrow windows (borrowed from inspector - though I don't know if I implemented this optimally, it seems to work)
- Drop the prefix from the docs link in the footer at narrow widths

The only significant section that's bad for narrow panes is the Debugger screen; that one isn't just a simple change!

It all seems good to me, but it'd be good to have another set of eyes play around with it (in addition to reviewing the code) to ensure it seems reasonable (including the points at which things start collapsing).

@jacob314 I removed a workaround you added (setting width to 0 for logging details) because it was breaking the vertical view and I couldn't repro any issue without it (though the CSS is a little changed now). If you think this is a problem, let me know!

File: packages/devtools/lib/src/inspector/inspector.css [hunk 1]
```diff
@@ -15,24 +15,13 @@
...
-    display: flex;
     overflow: hidden;
     margin-top: 10px;
     align-items: center;
     flex: 1 1 auto;
-    flex-direction: column;
     box-sizing: border-box;
 }
 
-/* Minimum aspect ratio */
```
Comment (jacob314) [inline packages/devtools/lib/src/inspector/inspector.css]: why is this removed? This is needed to make the inspector switch between vertical and horizontal as the the window is resized.

File: packages/devtools/lib/src/inspector/inspector.css [hunk 2]
```diff
@@ -47,32 +36,6 @@
     margin-left: 25px;
 }
 
-/*
```
Comment (jacob314) [inline packages/devtools/lib/src/inspector/inspector.css]: I would add these back but add an extra css tag.
Essentially these buttons are "collapse-early" 
We need to collapse them at width 1220px but there are other buttons we can wait to collapse for a while.

Comment (jacob314) [reply packages/devtools/lib/src/inspector/inspector.css]: never mind. looks like you added some nice collapse at various size options so the same effect is achieved

File: packages/devtools/lib/src/inspector/inspector.css (hunk 1 above)
Comment (jacob314) [reply packages/devtools/lib/src/inspector/inspector.css]: nm. was moved not removed.

File: packages/devtools/web/styles.css [hunk 3]
```diff
@@ -799,3 +824,69 @@ html [full] {
...
+
+/*
+ * Support tagging elements with collapsible that can have children tagged
+ * with optional-text to be hidden when the page is below a certain threshold.
+ *
+ * Also support optional directly on elements to be hidden.
+ */
+
+@media all and (max-width: 550px) { .optional-550 { display: none; } }
+@media all and (max-width: 700px) { .optional-700 { display: none; } }
```
Comment (jacob314) [inline packages/devtools/web/styles.css]: nit: maybe put all the CSS that depend on a specific max-width together in the same media query. So add the optional-700 and collapsible-700 rules to the same media query.

Comment (DanTup) [reply packages/devtools/web/styles.css]: I had it that way originally, but it was harder to glance in the CSS to see which widths we already had classes for that you could use, and which you needed to add new ones for (it kinds bugs me that we have to add a rule for each one, but I don't know there's a better way).

WDYT?

Comment (jacob314) [reply packages/devtools/web/styles.css]: Go ahead and leave it as is.

**Comment by jacob314** [comment]
lgtm

**Comment by DanTup** [comment]
Thanks - I've made a note of the other things you noted above and will follow up on them this week.
